src/components/App.js

```diff
@@ -12,21 +12,42 @@ import QueryAuth0 from 'console/components/data/QueryAuth0';
 import Routes from 'console/components/Routes';
 import CircleLogo from 'console/components/svg/CircleLogo';
 import { getError } from 'console/state/auth/selectors';
+import { getCurrentRouteTree } from 'console/state/router/selectors';
 import { reverse } from 'console/urls';
 
 const { Header } = Layout;
 
 @connect((state, props) => ({
   authError: getError(state),
+  routeTree: getCurrentRouteTree(state),
 }))
 export default class App extends React.Component {
   static propTypes = {
     authError: PropTypes.object,
     history: PropTypes.object.isRequired,
+    routeTree: PropTypes.array.isRequired,
   };
 
-  componentDidUpdate() {
-    const { authError } = this.props;
+  updateDocumentTitle() {
+    const { routeTree } = this.props;
+    const { documentTitle } = [...routeTree][0];
```

Comment:
You're not mutating the `routeTree` here, are you? So maybe there's no need for the hack-y spread operator trick here?

Reply:
oops! good catch. I did originally have to reverse this because it had been mutated. this is no longer necessary.